internal/modulespecifiers/specifiers.go

```diff
@@ -644,6 +659,54 @@ func tryGetModuleNameAsNodeModule(
 	overrideMode core.ResolutionMode,
 ) string {
 	parts := getNodeModulePathParts(pathObj.FileName)
+
+	// TODO understand what feature this part impacts
```

Comment:
It's the autoimport - when you're adding a new symbol into the file, TS will suggest you to import it from another file. To do that it'll map that other file path into an import path, and that's the function responsible for that.